File: data/readable_format/fortran/file2071.f90
MODULE ExternalInterface
    IMPLICIT NONE

    LOGICAL, PUBLIC :: haveExternalInterfaceBCVTB = .FALSE.
    INTEGER, PUBLIC :: NumExternalInterfaces = 0

CONTAINS

    SUBROUTINE ExternalInterfaceExchangeVariables
        IMPLICIT NONE
        LOGICAL, SAVE :: GetInputFlag = .TRUE.

        IF (GetInputFlag) THEN
            NumExternalInterfaces = GetNumObjectsFound('ExternalInterface')
            GetInputFlag = .FALSE.
            IF (NumExternalInterfaces > 0) THEN
                CALL ShowFatalError('ExternalInterface is not available in this version.')
            END IF
        END IF
    END SUBROUTINE ExternalInterfaceExchangeVariables

    SUBROUTINE CloseSocket(FlagToWriteToSocket)
        INTEGER, INTENT(IN) :: FlagToWriteToSocket
        ! This subroutine is intentionally left empty for this example
    END SUBROUTINE CloseSocket

    FUNCTION GetNumObjectsFound(objectType) RESULT(num)
        CHARACTER(len=*), INTENT(IN) :: objectType
        INTEGER :: num

        ! Simulate the behavior of finding objects.
        num = 0
        ! For testing, you can adjust the number returned to simulate different scenarios.
    END FUNCTION GetNumObjectsFound

    SUBROUTINE ShowFatalError(errorMessage)
        CHARACTER(len=*), INTENT(IN) :: errorMessage
        PRINT *, "Fatal Error:", errorMessage
    END SUBROUTINE ShowFatalError

END MODULE ExternalInterface

PROGRAM TestExternalInterface
    USE ExternalInterface
    IMPLICIT NONE

    CALL ExternalInterfaceExchangeVariables()

    PRINT *, "Test for ExternalInterfaceExchangeVariables passed."
END PROGRAM TestExternalInterface

File: data/readable_format/cpp/file2071.cpp
#include <iostream>
#include <string>

// Assuming the existence of these functions or simulating them
int GetNumObjectsFound(const std::string& objectType) {
    // Simulate finding no objects for simplicity
    return 0;
}

void ShowFatalError(const std::string& errorMessage) {
    std::cerr << "ShowFatalError: " << errorMessage << std::endl;
    exit(1); // Exit the program indicating failure, as a real fatal error would.
}

class ExternalInterface {
private:
    bool haveExternalInterfaceBCVTB = false;
    static bool GetInputFlag;

public:
    static int NumExternalInterfaces;

    ExternalInterface() {}

    void ExternalInterfaceExchangeVariables() {
        if (GetInputFlag) {
            NumExternalInterfaces = GetNumObjectsFound("ExternalInterface");
            GetInputFlag = false;
            if (NumExternalInterfaces > 0) {
                ShowFatalError("ExternalInterface is not available in this version.");
            }
        }
    }

    void CloseSocket(int FlagToWriteToSocket) {
        // Placeholder: No operation defined for closing socket in the simplified example
    }

    ~ExternalInterface() {}
};

bool ExternalInterface::GetInputFlag = true;
int ExternalInterface::NumExternalInterfaces = 0;

int main() {
    ExternalInterface ei;
    ei.ExternalInterfaceExchangeVariables();

    // Basic assertion to simulate a simple test case
    if (ExternalInterface::NumExternalInterfaces == 0) {
        std::cout << "Test passed: No external interfaces found as expected." << std::endl;
    } else {
        std::cerr << "Test failed: Unexpected number of external interfaces found." << std::endl;
        return 1; // Indicate failure
    }

    // Add any additional tests or operations here

    return 0; // Indicate success
}